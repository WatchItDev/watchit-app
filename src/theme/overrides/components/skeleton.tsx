import { Theme } from '@mui/material/styles';

// ----------------------------------------------------------------------

export function skeleton(theme: Theme) {
  return {
    MuiSkeleton: {
      styleOverrides: {
        root: {
          backgroundColor: theme.palette.background.neutral,
        },
        rounded: {
          borderRadius: theme.shape.borderRadius * 2,
        },
      },
    },
  };
}
