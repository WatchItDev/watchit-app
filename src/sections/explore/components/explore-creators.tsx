import CarouselCreators from '@src/components/carousel/variants/carousel-creators.tsx';
import { filterHiddenProfiles } from '@src/libs/profile.ts';
import { useGetRecentUsersQuery } from '@src/graphql/generated/hooks.tsx';
import { User } from '@src/graphql/generated/graphql.ts';
import { ExploreCreatorsSkeleton } from '@src/sections/explore/components/explore-creators.skeleton.tsx';
import { LoadingFade } from '@src/components/LoadingFade.tsx';

// ----------------------------------------------------------------------

export const ExploreCreators = () => {
  const { data, loading } = useGetRecentUsersQuery({ variables: { limit: 50 } })

  // FilteredCompletedProfiles is an array of objects, each object has a metadata property and inside exists a displayName en bio property; filter the profiles that not have a displayName and bio property
  const filtered = data?.getRecentUsers?.filter(
    (profile: User) => profile?.displayName && profile?.bio
  );

  // Clear ###HIDDEN### profiles
  const filteredProfiles = filterHiddenProfiles(filtered);

  return (
    <LoadingFade loading={loading} skeleton={<ExploreCreatorsSkeleton />} delayMs={500}>
      {!!filteredProfiles?.length && (
        <CarouselCreators
          profiles={filteredProfiles}
          title="Latest creators"
          minItemWidth={250}
          maxItemWidth={400}
        />
      )}
    </LoadingFade>
  );
}
