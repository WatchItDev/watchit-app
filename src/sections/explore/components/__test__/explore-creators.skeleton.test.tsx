import {describe,expect,it} from 'vitest'
import {render} from '@testing-library/react'
import { ExploreCreatorsSkeleton } from '../explore-creators.skeleton';

describe('Testing in the <ExploreCreatorsSkeleton/> ', () => {
  it('should render the skeleton and match snapshot', () => {
    const { container } = render(<ExploreCreatorsSkeleton/>);
    expect(container).toMatchSnapshot();
  });
})

