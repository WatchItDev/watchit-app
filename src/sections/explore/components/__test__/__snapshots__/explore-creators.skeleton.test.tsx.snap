// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing in the <ExploreCreatorsSkeleton/>  > should render the skeleton and match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-19i2hl5-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardHeader-root css-hl0h5w-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
        >
          Latest creators
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root css-ceyv38-MuiCardContent-root"
    >
      <div
        class="MuiBox-root css-i6bazn"
      >
        <div
          class="MuiBox-root css-k008qs"
        >
          <div
            class="MuiBox-root css-rmtq2p"
          >
            <div
              class="MuiBox-root css-1ml1j7v"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-18udj95-MuiSkeleton-root"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1d1bhsb-MuiSkeleton-root"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1c88oq8-MuiSkeleton-root"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1yige66-MuiSkeleton-root"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-k008qs"
        >
          <div
            class="MuiBox-root css-rmtq2p"
          >
            <div
              class="MuiBox-root css-1ml1j7v"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-18udj95-MuiSkeleton-root"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1d1bhsb-MuiSkeleton-root"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1c88oq8-MuiSkeleton-root"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1yige66-MuiSkeleton-root"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
